projects/extension/src/containers/Popup.tsx

```diff
@@ -20,27 +26,62 @@ const SearchBar: React.FC = (): React.ReactElement => (
 );
 
 const Popup: React.FunctionComponent = () => {
+  const [activeTab, setActiveTab] = React.useState<React.ReactElement | undefined>();
+  const [rTabs, setRTabs] = React.useState<React.ReactElement[]>([]);
   const appliedTheme = createMuiTheme(light);
-  const tabs = useTabs();
-  /* this is kinda of how  the access is to chrome tabs */
-  chrome.tabs.query({'active': true, 'windowId': chrome.windows.WINDOW_ID_CURRENT}, 
-  (tabs) => {
-    console.log('tabs', tabs, tabs[0].url)
-  });
-  const tabCurrent = tabs[0];
-  const tabsRest = tabs.slice(1);
+  const [apps, setApps] = React.useState<AppMediator[]>([]);
+
+  React.useEffect((): void => {
+    chrome.runtime.getBackgroundPage(backgroundPage => {
+      const bg = backgroundPage as Background;
+      bg.manager && setApps(bg.manager.apps);
+
+      // TODO: change the state when stateChanged occurs
```

Comment:
I think this is as simple as moving `bg.manager && setApps(bg.manager.apps);` inside the event handler

Reply:
Please elaborate

Reply:
It is a bit more complicated as the state change should alter the state of the app in the existing apps array and reset - so I will leave that for a later PR.